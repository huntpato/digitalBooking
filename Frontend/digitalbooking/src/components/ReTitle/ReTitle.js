import React from "react";
import styles from "./reTitle.module.css";

export const ReTitle = (props) => {
  const { type, title } = props;
  const { titles } = styles;

  return (
    <div className={titles}>
      <p>{type}</p>
      <h5>{title}</h5>
    </div>
  );
};
