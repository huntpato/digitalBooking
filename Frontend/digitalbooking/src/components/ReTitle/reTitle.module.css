.titles{
    display: flex;
    flex-direction: column;
    width:75%;
}

.titles h5{
    font-size: 18px;
    overflow:hidden;
    padding-top:2px ;
    text-overflow:ellipsis;    
    white-space:nowrap;
    width: 100%;
}

.titles p{
    font-size: 11px;
    
}

@media (max-width: 480px){
    .titles p{
        padding-top: 0px;
    }

}