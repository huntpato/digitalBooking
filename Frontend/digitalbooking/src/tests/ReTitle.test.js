import React from 'react';
import { shallow } from 'enzyme';
import { ReTitle } from '../components/ReTitle/ReTitle'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <ReTitle />', ()=> {

    const type = "Hotel";
    const title = "Hotel Hermitage";
    const wrapper= shallow(<ReTitle type={type} title={title}/>);

    test('Comprobar que ReTitle renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Comprobar que se renderiza el tipo de propiedad enviado por props', () => {
        const p = wrapper.find('p').text();
        expect( p ).toContain( type );
    }) 

    
    test('Comprobar que se renderiza el titulo de la propiedad enviado por props', () => {
        const h5 = wrapper.find('h5').text();
        expect( h5 ).toContain( title );
    }) 

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const div = wrapper.find('div');
        expect(div.hasClass('titles')).toBe(true);
    })
})
