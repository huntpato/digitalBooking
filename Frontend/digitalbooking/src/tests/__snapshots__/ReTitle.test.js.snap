// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <ReTitle /> Comprobar que ReTitle renderiza correctamente. 1`] = `
<div
  className="titles"
>
  <p>
    Hotel
  </p>
  <h5>
    Hotel Hermitage
  </h5>
</div>
`;
